### PR TITLE
Fixes #7 Added setup instruction to build the project

### LINKED ISSUE #7: Add a "SETUP" part to readme.md

### PR DESCRIPTION
**Pull request by cyberhck**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,6 +10,30 @@ Main customer's will be BaseApi classes in...
...
+If you don't have npm installed, please do it.
+
+If you get permission warnings, you should really fix it the right way, but if you want, you can use dirty way (`sudo`).
+
+Take a look at [npm docs](https://docs.npmjs.com/getting-started/fixing-npm-permissions) to fix the right way.
+
+Now to install dependencies, just execute the following commands.
+- `npm install` (installs npm dependencies; mostly for development)
+- `jspm install` (For typescript and few polyfill; mostly for development)
+- `tsd reinstall` (to install typings provided by jasmine)
```
Comment (wmathes) [inline README.md]: => typing install

File: README.md [hunk 2]
```diff
@@ -10,6 +10,30 @@ Main customer's will be BaseApi classes in...
...
+- `npm run compile` (compiles everything)
+
+## Running tests
+Just execute `npm run compile && npm run test && gulp tslint` and it should compile, test, and run a lint.
```
Comment (wmathes) [inline README.md]: "gulp tslint" should be "npm run lint"

File: README.md [hunk 3]
```diff
@@ -10,6 +10,30 @@ Main customer's will be BaseApi classes in...
...
+
+If you get permission warnings, you should really fix it the right way, but if you want, you can use dirty way (`sudo`).
+
+Take a look at [npm docs](https://docs.npmjs.com/getting-started/fixing-npm-permissions) to fix the right way.
+
+Now to install dependencies, just execute the following commands.
+- `npm install` (installs npm dependencies; mostly for development)
+- `jspm install` (For typescript and few polyfill; mostly for development)
+- `tsd reinstall` (to install typings provided by jasmine)
+- `npm run compile` (compiles everything)
```
Comment (wmathes) [inline README.md]: Not part of "Setup"

File: README.md (hunk 2 above)
Comment (wmathes) [inline README.md]: Split in lines, Explain.

Comment (wmathes) [inline README.md]: Add ## Build section

File: README.md (hunk 3 above)
Comment (cyberhck) [inline README.md]: Removed from here and is described in testing section

File: README.md (hunk 2 above)
Comment (cyberhck) [inline README.md]: done `npm run lint`
split into lines explaining
and also a one line command

Comment (cyberhck) [inline README.md]: added a build section with just `npm run build` and explain what it does.

**Pull request merged**